test/rql_test/connections/twisted_connection.py

```diff
@@ -0,0 +1,950 @@
+        self._run(result)
+
+    def setUp(self):
+        return defer.succeed(None)
+
+    def tearDown(self):
+        return defer.succeed(None)
+
+    @inlineCallbacks
+    def _run(self, result=None):
+        if result is None:
+            result = self.defaultTestResult()
+        result.startTest(self)
+        testMethod = getattr(self, self._testMethodName)
+
+        try:
+            try:
+                yield self.setUp()
+            except KeyboardInterrupt:
+                raise
+            except:
+                result.addError(self, sys.exc_info())
+                return
+
+            ok = False
+            try:
+                yield testMethod()
+                ok = True
+            except KeyboardInterrupt:
+                raise
+            except Failure:
+                result.addFailure(self, sys.exc_info())
+            except:
+                result.addError(self, sys.exc_info())
+
+            try:
+                yield self.tearDown()
+            except KeyboardInterrupt:
+                raise
+            except:
+                result.addError(self, sys.exc_info())
+                ok = False
+            if ok:
+                result.addSuccess(self)
+        finally:
+            result.stopTest(self)
+
+
+class TestWithConnection(TestCaseCompatible):
+    port = None
+    server = None
+    serverOutput = None
+    ioloop = None
+
+    @inlineCallbacks
+    def setUp(self):
+        global sharedServer, sharedServerOutput, sharedServerHost, \
+            sharedServerDriverPort
+
+        if sharedServer is not None:
+            try:
+                yield sharedServer.check()
+            except Exception:
+                # TODO: figure out how to blame the last test
+                closeSharedServer()
+
+        if sharedServerDriverPort is None:
+            sharedServerOutput = tempfile.NamedTemporaryFile('w+')
+            sharedServer = driver.Process(executable_path=rethinkdb_exe,
+                                          console_output=sharedServerOutput,
+                                          wait_until_ready=True)
+            sharedServerHost = sharedServer.host
+            sharedServerDriverPort = sharedServer.driver_port
+
+        # - ensure we are ready
+
+        yield checkSharedServer()
+
+    @inlineCallbacks
+    def tearDown(self):
+        global sharedServer, sharedServerOutput, sharedServerHost, \
+            sharedServerDriverPort
+
+        if sharedServerDriverPort is not None:
+            try:
+                yield checkSharedServer()
+            except Exception:
+                closeSharedServer()
+                raise  # TODO: figure out how to best give the server log
+
+# == Test Classes
+
+
+class TestNoConnection(TestCaseCompatible):
+    # No servers started yet so this should fail
+    @inlineCallbacks
+    def test_connect(self):
+        if not use_default_port:
+            return None  # skipTest will not raise in this environment
```

Comment:
This line is failing when I try to run the tests:

`SyntaxError: 'return' with argument inside generator`